src/coreclr/src/tools/Common/TypeSystem/IL/Stubs/RuntimeHelpersIntrinsics.cs

```diff
@@ -30,6 +30,19 @@ public static MethodIL EmitIL(MethodDesc method)
                 return emit.Link(method);
             }
 
+            if (methodName == "GetMethodTable")
+            {
+                ILEmitter emit = new ILEmitter();
+                ILCodeStream codeStream = emit.NewCodeStream();
+                codeStream.EmitLdArg(0);
+                codeStream.Emit(ILOpcode.ldflda, emit.NewToken(method.Context.SystemModule.GetKnownType("System.Runtime.CompilerServices", "RawData").GetField("Data")));
+                codeStream.EmitLdc(-method.Context.Target.PointerSize);
```

Comment:
I didn't see any cleaner way to inject `-IntPtr.Size` as a constant. Let me know if there's a preferred alternative.

Reply:
This is the right way to do this.